lib/resque/scheduler/delaying_extensions.rb

```diff
@@ -212,22 +204,21 @@ def find_delayed_selection(klass = nil, &block)
       def remove_delayed_job_from_timestamp(timestamp, klass, *args)
         return 0 if Resque.inline?
 
-        key = "delayed:#{timestamp.to_i}"
-        encoded_job = encode(job_to_hash(klass, args))
+        job_hash = job_to_hash(klass, args)
+
+        redis.srem("timestamps:#{encode(job_hash)}", "delayed:#{timestamp.to_i}")
 
-        redis.srem("timestamps:#{encoded_job}", key)
-        count = redis.lrem(key, 0, encoded_job)
-        clean_up_timestamp(key, timestamp)
+        ret = redis.zrangebyscore(:delayed_queue, timestamp.to_i, timestamp.to_i, with_scores: true).map do |job, time|
+          next unless time == timestamp.to_i
```

Comment:
Why do we need this? Doesn't Redis take care of filtering elements with `time != timestamp`?

Reply:
Its due to floating point nonsense.